samples/winapi_drivelist/main.c

```diff
@@ -9,45 +9,46 @@ int main(void)
 
     // Mount some drives for demonstration purposes
     BOOL ret;
+    DWORD error;
     ret = nxMountDrive('C', "\\Device\\Harddisk0\\Partition2\\");
     if (!ret) {
-        debugPrint("Failed to mount C: drive!\n");
-        Sleep(5000);
-        return 1;
+        // Additional error info can be retrieved with GetLastError()
+        error = GetLastError();
+        debugPrint("Failed to mount C: drive! Error code: %x\n", error);
+        goto end;
     }
+
     ret = nxMountDrive('E', "\\Device\\Harddisk0\\Partition1\\");
     if (!ret) {
-        debugPrint("Failed to mount E: drive!\n");
-        Sleep(5000);
-        return 1;
+        error = GetLastError();
+        debugPrint("Failed to mount E: drive! Error code: %x\n", error);
+        goto unmount_c;
```

Comment:
This code makes little sense to me; mounting E failed, so it jumps to unmounting C?
The labels should be named more appropriately (like `handleError` or `end`)

Reply:
I believe that this specific label should stay as unmount_c, it's a good enough name IMO.

Reply:
@JayFoxRox What would you suggest? Personally, it doesn't bother me, imho it looks just like when a function is freeing a previous allocation before returning due to an error.

Reply:
I'm not sure; I'd probably cause a fatal error if anything fails to init (to avoid having to do conditional cleanup).

If conditional cleanup is mandatory I'd probably have a boolean which stores which drives were mounted correctly (`mountedC` / `mountedE`), I'd then have a generic cleanup function at the end of the scope which unmounts it (a single label called `cleanup`, which would conditionally deal with cleanup).

In a real application I'd probably have something like `atexit` / destructors which deal with this (= no labels); also ensures proper order of destruction without having to think about it.
Probably not practical for a sample because it adds too much complexity.

Reply:
Especially with everything else jumping to `cleanup` now, I think this should use booleans / keep track of it and handle everything in `cleanup`